plugins/module_utils/parsing.py

```diff
@@ -11,7 +11,7 @@
 METRIC_SUFFIXES = ["K", "M", "G", "T", "P"]
 
 duration_pattern = re.compile(
-    r"^((?P<Y>[1-9]\d*)Y)?((?P<W>[1-9]\d*)W)?((?P<D>[1-9]\d*)D)?(((?P<H>[1-9]\d*)H)?((?P<M>[1-9]\d*)M)?)?$"
+    r"^0*((?P<Y>[1-9]\d*)Y)?((?P<W>[1-9]\d*)W)?((?P<D>[1-9]\d*)D)?(((?P<H>[1-9]\d*)H)?((?P<M>[1-9]\d*)M)?)?$"
```

Comment:
You should allow leading zeros for each group. You should just change  `[1-9]\d*` in all groups to `\d+`

Reply:
Did not read Jan's comment. I will resolve this conversation as a duplicate